### PR TITLE
Curricular management API cleanup and enhancement

### PR DESCRIPTION
**Pull request by NunoPinheiro**
Closes [ACADEMIC-638](https://jira.fenixedu.org/browse/ACADEMIC-638)

### CODE REVIEW

**Comment by NunoPinheiro** [comment]
also closes ACADEMIC-630

File: src/main/java/org/fenixedu/academic/domain/exceptions/DomainException.java [hunk 1]
```diff
@@ -50,4 +50,5 @@ public static void throwWhenDeleteBlocked(Collection<String> blockers) {
             throw new DomainException("key.return.argument", blockers.stream().collect(Collectors.joining(", ")));
         }
     }
+
```
Comment (jcarvalho) [inline src/main/java/org/fenixedu/academic/domain/exceptions/DomainException.java]: Please avoid these types of changes...

**Comment by nurv** [comment]
@NunoPinheiro "temp" is not a proper commit message. Be careful with this.

**Comment by jcarvalho** [comment]
Your code could be much simpler if Java 8 features/APIs were used, but otherwise looks good.

**Comment by pedrosan7os** [comment]
https://github.com/FenixEdu/fenixedu-academic/pull/645 is in conflict with this one, please rebase

**Pull request merged**